### PR TITLE
backport another patch relating to new gcc versions

### PR DESCRIPTION
**Pull request by dwightguth**
Tested on Arch Linux with gcc 8.1.

Ready for review.

### CODE REVIEW

File: k-distribution/src/main/scripts/lib/opam/compilers/4.03.0/4.03.0+k/files/opam-compiler.patch [hunk 1]
```diff
@@ -20,6 +20,37 @@ index 7d54160..5a51708 100644
...
+     sock_domain = PF_UNIX;
+     sock_addr.s_unix.sun_family = AF_UNIX;
++    a_len = strlen(address);
++    if (a_len >= sizeof(sock_addr.s_unix.sun_path)) {
++      caml_fatal_error("Debug socket path length exceeds maximum permitted length");
++    }
+     strncpy(sock_addr.s_unix.sun_path, address,
+-            sizeof(sock_addr.s_unix.sun_path));
++            sizeof(sock_addr.s_unix.sun_path) - 1);
++    sock_addr.s_unix.sun_path[sizeof(sock_addr.s_unix.sun_path) - 1] = '\0';
```
Comment (traiansf) [inline k-distribution/src/main/scripts/lib/opam/compilers/4.03.0/4.03.0+k/files/opam-compiler.patch:46]: If you fail with `fatal_error` above, why is this change still needed? Is  this just for safety, in case the above is just a warning?

**Review by traiansf — APPROVED** [review]

File: k-distribution/src/main/scripts/lib/opam/compilers/4.03.0/4.03.0+k/files/opam-compiler.patch (hunk 1 above)
Comment (dwightguth) [reply k-distribution/src/main/scripts/lib/opam/compilers/4.03.0/4.03.0+k/files/opam-compiler.patch:46]: It's not really my code, I'm just backporting a change from the trunk of ocaml.
